packages/gestalt/src/Toast.js

```diff
@@ -112,69 +116,89 @@ export default function Toast({
+          ) : null}
+          {isErrorVariant ? (
+            <Flex.Item flex="none">
+              <Icon
+                color="inverse"
+                icon="workflow-status-problem"
+                accessibilityLabel="problem"
+                size={EXP_SIZE_ICON}
+              />
+            </Flex.Item>
+          ) : null}
+          <Flex.Item flex="grow">
+            <Text
+              weight={isErrorVariant ? 'bold' : undefined}
+              align="start"
+              color={textColor}
+              lineClamp={2}
+            >
+              {textElement}
+            </Text>
+          </Flex.Item>
 
-            {primaryAction || _dangerouslySetPrimaryAction ? (
-              // Allow button text to wrap on mobile
-              <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
-                {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
-                {!_dangerouslySetPrimaryAction &&
-                primaryAction?.accessibilityLabel &&
-                primaryAction?.label ? (
-                  <ToastPrimaryAction
-                    accessibilityLabel={primaryAction.accessibilityLabel}
-                    href={primaryAction.href}
-                    rel={primaryAction?.rel}
-                    size="sm"
-                    target={primaryAction?.target}
-                    label={primaryAction.label}
-                    onClick={primaryAction.onClick}
-                  />
-                ) : null}
-              </Flex.Item>
-            ) : null}
-          </Flex>
-        </Box>
+          {primaryAction || _dangerouslySetPrimaryAction ? (
+            // Allow button text to wrap on mobile
+            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+              {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
+              {!_dangerouslySetPrimaryAction &&
+              primaryAction?.accessibilityLabel &&
+              primaryAction?.label ? (
+                <ToastPrimaryAction
+                  accessibilityLabel={primaryAction.accessibilityLabel}
+                  href={primaryAction.href}
+                  rel={primaryAction?.rel}
+                  size="sm"
+                  target={primaryAction?.target}
+                  label={primaryAction.label}
+                  onClick={primaryAction.onClick}
+                />
+              ) : null}
+            </Flex.Item>
+          ) : null}
+        </Flex>
       </Box>
     );
+    return (
+      <Fragment>
+        <Box
+          marginEnd={8}
+          marginStart={8}
+          maxWidth={`min(${EXP_TOAST_MAX_WIDTH}px, 100vw)`} // Ensure that maxWidth isn't greater than viewport width (for small screens)
+          role="status"
+          rounding={4}
+          smDisplay="none"
+          width="100%" // only full width below the smallest breakpoint
+        >
+          {toastContent}
+        </Box>
+        <Box
```

Comment:
Alberto's right, this is how the responsive `display` props are intended to be used. I've never particularly liked it. 😅 But it does seem like the best solution here.

To dedupe props: though we lint against it in Pinboard, I would create a `sharedProps` object and spread it over Boxes. That does make future codemods slightly less efficient (upgrades here will have to be made manually), but avoids drift between the two Boxes, which seems like the more likely source of bugs.

Reply:
Just moved everything to a div. 

margins were simmetric so no concerns re RTL